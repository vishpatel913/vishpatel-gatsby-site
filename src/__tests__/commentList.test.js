import React from "react";
import renderWithTheme from "../utils/test-renderer";
import CommentList from "../components/commentList";

describe("CommentList", () => {
  let mockData;

  beforeEach(() => {
    mockData = [
      {
        node: {
          name: "tester",
          message: {
            message: "testing"
          },
          timestamp: 1519308700
        }
      }
    ];
  });

  it("renders correctly without data", () => {
    const tree = renderWithTheme(<CommentList data={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with data", () => {
    const tree = renderWithTheme(<CommentList data={mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
