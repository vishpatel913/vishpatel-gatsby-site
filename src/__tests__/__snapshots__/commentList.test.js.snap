// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList renders correctly with data 1`] = `
.c0 {
  display: block;
  margin: 0;
  margin-bottom: 2rem;
}

.c1 {
  margin: 0;
}

.c2 {
  display: block;
  margin-bottom: 1.5rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-left: 4px solid #512DA8;
  padding-left: 1em;
}

.c5 {
  display: block;
}

<div
  className="c0"
>
  <h2>
    1 
    Comment
  </h2>
  <ul
    className="c1"
  >
    <li
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <strong>
            Tester
          </strong>
          22 Feb '18
        </div>
      </div>
      <div
        className="c5"
      >
        testing
      </div>
    </li>
  </ul>
</div>
`;

exports[`CommentList renders correctly without data 1`] = `
.c0 {
  display: block;
  margin: 0;
  margin-bottom: 2rem;
}

.c1 {
  margin: 0;
}

<div
  className="c0"
>
  <h2>
    0 
    Comment
    s
  </h2>
  <ul
    className="c1"
  />
</div>
`;
